pkg/ccl/backupccl/restore.go

```diff
@@ -1449,6 +1452,17 @@ func (r *restoreResumer) OnSuccess(ctx context.Context, txn *client.Txn, job *jo
 		return errors.Wrapf(err, "restoring %d TableDescriptors", len(r.tables))
 	}
 
+	// Initiate a run of CREATE STATISTICS. We don't know the actual number of
+	// rows affected per table, but the stats refresher will always create
+	// statistics if none exist yet.
+	for i := range r.tables {
+		r.statsRefresher.NotifyMutation(
```

Comment:
The txn is not committed until some time after this OnSuccess func returns, which means the descriptors might not exist if the stats refresher decides to run before the txn is committed. What happens if the stats refresher attempts to fetch the descriptor for a table that doesn't exist? Will it fail? Retry later? If it fails and doesn't retry then a read-only import'd table does have a chance of never triggering a stats run.

Reply:
I would also recommend running this under stress on your machine to attempt to trigger this case, unless there's some txn or retry interaction I'm not aware of that makes this safe and non-racey.